### PR TITLE
fix possible crash in agent

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

File: assignment-client/src/Agent.cpp [hunk 1]
```diff
@@ -364,10 +364,14 @@ void Agent::processAgentAvatarAndAudio(float deltaTime) {
 }
 
 void Agent::aboutToFinish() {
-    _scriptEngine->stop();
+    if (_scriptEngine) {
```
Comment (ZappoMan) [inline assignment-client/src/Agent.cpp]: it's theoretically possible that the ThreadedAssignment could stop before the Agent::run() completed and created a _scriptEngine or _pingTimer

File: assignment-client/src/Agent.cpp [hunk 2]
```diff
@@ -364,10 +364,14 @@ void Agent::processAgentAvatarAndAudio(float deltaTime) {
...
+        _scriptEngine->stop();
+    }
 
-    _pingTimer->stop();
-    delete _pingTimer;
+    if (_pingTimer) {
+        _pingTimer->stop();
+        delete _pingTimer;
```
Comment (huffman) [inline assignment-client/src/Agent.cpp]: Should `_pingTimer` be set to `nullptr` after deleting? I doubt it would cause a problem either way.

Comment (ZappoMan) [inline assignment-client/src/Agent.cpp]: I think it's safe as is. this is essentially the last thing that gets done before shutting down.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 34dc8bdbcb8de07f6e6726efefd9462c67f241d5 available at: https://gist.github.com/a1b5e6104780d3272aea<br/><br/>No installer was built for this commit

**Pull request merged**